### ISSUE TITLE
Vue multiselect inside of element with scrollbar

### ISSUE THREAD

**Issue by MattCCC** [shentao/vue-multiselect#723]
Hi, firstly thank you for a great plugin, it's really great.

I found one issue with using vue vue-multiselect inside of elements with `overflow:auto`. After to expand, vue multiselect is still inside of the element with overflow auto, which is not how the standard `<select>` works.

Fiddle: [https://jsfiddle.net/jqofkzxc/3078/](https://jsfiddle.net/jqofkzxc/3078/)

Expected behaviour: After to expand the vue-multiselect (`.multiselect__content-wrapper`) shouldn't cause the `.modal-body` to expand.

Possible fix: This could be achieved by setting it outside of element with overflow (.modal-body in that case), and using position absolute + dynamically counted left/top css.

Why? It would make it much easier to use it in replacement to standard <select> without any hacks.

**Comment by shentao** [comment]
Yeah, that’s an issue related to the current structure of the code – the dropdown is located just next to the input, thus it is also affected by all the overflows of the code around it.

I think you can try to fix this with some combination of position `absolute` and `static`. Here’s a nice article about that: https://css-tricks.com/popping-hidden-overflow/

**Comment by me-singh** [comment]
Is there any way we can overcome this issue. I want dropdown to appear above overflow (it is blocked by overflow).

**Comment by sigma207** [comment]
@me-singh 
You can try this [directive](https://gist.github.com/sigma207/b9300fe12a996c07b2389ee03c1464ed).

**Comment by Kocal** [comment]
I've used @sigma207 directive, but unfortunately I've got two issues with it:
- when scrolling, the multiselect does not move and that's weird
- the parent element lose its height since the multiselect is in `position: fixed`, so next elements are moved up and this is not wanted
 
I've patched it to use `position: absolute` instead of `fixed`, and used an invisible clone to preserve the parent height:

```js
export default {
  install(Vue) {
    Vue.directive('select-overflow', {
      inserted: (el, _binding, vnode) => {
        let originalWidth
        let originalPosition
        let originalZIndex
        let selectIsOpen = false
  
        // will be used as a placeholder, in order to fix parent height issue (due to position relative)
        const clone = el.cloneNode(true);
        clone.style.visibility = 'hidden';

        vnode.child.$watch('isOpen', isOpen => {
          selectIsOpen = isOpen
          if (isOpen) {
            const { offsetWidth } = el
            originalWidth = el.style.width
            originalPosition = el.style.position
            originalZIndex = el.style.zIndex
            el.style.width = `${offsetWidth}px`
            el.style.position = 'absolute'
            el.style.zIndex = 2
            el.parentNode.insertBefore(clone, el.nextSibling); // insert after el
          } else {
            el.style.position = originalPosition
            el.style.width = originalWidth
            el.style.zIndex = originalZIndex
            clone.parentNode.removeChild(clone);
          }
        })

        window.addEventListener('wheel', event => {
          if (selectIsOpen) {
            // disabled outside scroll when select is open
            event.stopPropagation()
          }
        }, true)
      },
    })
  }
}
```

**Comment by ryzhak-andrii** [comment]
@Kocal It work's only for the not scrollable parent with a select within. In another case, I have this result 
![Screen Shot 2020-07-20 at 6 30 15 PM](https://user-images.githubusercontent.com/55483532/87956132-55786180-cab7-11ea-9703-1059db620097.png)

**Comment by Kocal** [comment]
@ryzhak-andrii I think I had the same issue than you, can you try the new snippet in https://github.com/shentao/vue-multiselect/issues/723#issuecomment-660342650? I'm now setting `position: relative` to the vue-multiselect parent.

**Comment by yasoza** [comment]
Here, did the fix for overflow or z-index issues by appending options to body with the help of@Kocal's comment and appending the options to body.
[PR](https://github.com/shentao/vue-multiselect/pull/1350)

**Comment by renanreis** [comment]
This workaround worked for me:

.multiselect__content-wrapper {
  position: static;
}

**Comment by atymic** [comment]
Anyone got this working on vue3?

**Comment by mattelen** [comment]
> Anyone got this working on vue3?

Trying installing the latest beta version as that supports Vue 3. If you are still having issues, please open a new issue

**Comment by atymic** [comment]
> > Anyone got this working on vue3?
> 
> Trying installing the latest beta version as that supports Vue 3. If you are still having issues, please open a new issue

More I meant the gists, they seem to work only with vue2
